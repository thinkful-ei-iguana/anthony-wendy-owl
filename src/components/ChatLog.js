import React from "react";
import ChatItem from "./ChatItem";
import "./ChatLog.css";

export default function ChatLog(props) {
  const cList = props.chatEvents.map(item => (
    <ChatItem
      key={item.participantId}
      type={item.type}
      message={item.message}
      time={item.time}
      timestamp={item.timestamp}
    />
  ));
  return (
    <div className="chat-panel">
      <h1>Chat</h1>
      <div className="c-list">
        <li>{cList}</li>
      </div>
    </div>
  );
}
ChatLog.defaultProps = {
  chatEvents: []
};
