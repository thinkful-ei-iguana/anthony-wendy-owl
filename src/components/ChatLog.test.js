import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ChatLog from "./ChatLog";

describe("ChatLog component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ChatLog />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the UI as expected", () => {
    const tree = renderer.create(<ChatLog message="test message" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
