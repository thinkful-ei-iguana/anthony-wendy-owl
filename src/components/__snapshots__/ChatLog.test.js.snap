// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatLog component renders the UI as expected 1`] = `
<div
  className="chat-panel"
>
  <h1>
    Chat
  </h1>
  <div
    className="c-list"
  >
    <li />
  </div>
</div>
`;
